### PR TITLE
Added machine constraint checker service

### LINKED ISSUE #30: Constraint checker service
Add support for handling the machine_constraints section

```yaml
machine_constraints:
  min_cores: 32
  min_memory: "20tb"
```

### PR DESCRIPTION
**Pull request by schaefi**
Validate min_cores and min_memory constraints on the
machine when booting. This Fixes #30 

__Note for myself__

- [x] add service to report for inclusion in total success validation #32

### CODE REVIEW

File: azure_li_services/runtime_config.py [hunk 1]
```diff
@@ -101,6 +101,10 @@ def get_network_config(self):
         if self.config_data and 'networking' in self.config_data:
             return self.config_data['networking']
 
+    def get_machine_constraints(self):
+        if self.config_data and 'machine_constraints' in self.config_data:
+            return self.config_data['machine_constraints']
```
Comment (rjschwei) [inline azure_li_services/runtime_config.py]: Skip the 'and' condition simply use:

return self.config_data.get('machine_constraints', None)

Similar for the others

File: azure_li_services/units/machine_constraints.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                'Number of cores: {0} is below required minimum: {1}'.format(
+                    existing_cores, min_cores
+                )
+            )
+
+
+def check_main_memory_validates_constraint(machine_constraints):
+    min_memory = machine_constraints.get('min_memory')
+    if min_memory:
+        min_bytes = humanfriendly.parse_size(min_memory)
```
Comment (jeffaco) [inline azure_li_services/units/machine_constraints.py]: How human friendly is `humanfriendly.parse_size(min_memory)`?

Would this accept something like 2.5TB, or would that be invalid? I'm wondering just what kind of input would be allowed here, thanks.

/Jeff

Comment (schaefi) [reply azure_li_services/units/machine_constraints.py]: sure it accepts almost everything. Here is an example and you can try it out yourself if you like

```python
import humanfriendly

print(humanfriendly.parse_size('1.5tb'))
print(humanfriendly.parse_size('2G'))
print(humanfriendly.parse_size('2g'))
print(humanfriendly.parse_size('2gb', binary=True))
print(humanfriendly.parse_size('0.24kb'))
```

It doesn't matter if you pass 'k, kb, K, KB, b, G, gb, GB ....' it's simply great :smile: 

by the way I realized by default it calculates with a power of ten which is not correct for dealing with storage sizes where we have bytes (power of two). Thus I have changed the call to use the binary option

**Comment by schaefi** [comment]
@jeffaco I guess I could clarify what size format you could use with the config. Thus I'll continue merging this even without you having approved as I think it's ok to go ahead. If there are still questions on the format of a size specification feel free to open an issue or mail. Thanks

**Pull request merged**